src/TodoList.js

```diff
@@ -1,41 +1,47 @@
 class TodoList {
-  constructor () {
+  constructor() {
     this.id = 0
     this.items = []
   }
 
-  create (str) {
+  create(str) {
     this.id++
     const item = { id: this.id, text: str, status: 'incomplete' }
     this.items.push(item)
     return item
   }
 
-  showAll () {
+  showAll() {
     return this.items
   }
 
-  setComplete (id) {
+  setComplete(id) {
     const item = this.findBy(id)
     item.status = 'complete'
     return item
   }
 
-  getByStatus (status) {
-    return this.items.filter(item => item.status === status)
+  getByStatus(status) {
+    return this.items.filter((item) => item.status === status)
   }
 
-  findBy (id) {
-    const item = this.items.find(item => item.id === id)
+  findBy(id) {
+    const item = this.items.find((item) => item.id === id)
     if (item === undefined) throw new Error('Item not found')
     return item
   }
 
-  deleteBy (id) {
+  deleteBy(id) {
     const item = this.findBy(id)
     const index = this.items.indexOf(item)
     return this.items.splice(index, 1)[0]
   }
+
+  showMaxChar() {
+    // I think i have the right idea in the rough
+    // but as i am faling i am certain my logic is not sound.
+    return TodoList.showAll.text.subtstring(0, 20) + '.....'
```

Comment:
You're on the right track with using the substring function but this line isn't right. Let's break each part down...

`TodoList` <- this is a class name
`.showAll` <- this is a function which returns an array
`.text` <- this is a property that exists on a single todo object

If you want to change the value of a property on an object, you do this:

```js
myObject.aProperty = 'some new value'
```

If you have an array of objects, you'll need to loop over the array to perform that logic on each object in the array:
```js
for (let i = 0; i < theArray.length; i++) {
  theArray[i].aProperty = 'some new value'
}
```

Alternatively, you can use the array `.map` function: 

```js
theArray.map(myObject => {
  myObject.aProperty = 'some new value'
  return myObject
})
```

Reply:
Thank you for the feedback once again! I can work off this